### PR TITLE
Add .all method to aggregate results over many pages

### PR DESCRIPTION
**Pull request by toddmohney**
Resolves: https://github.com/zephyr-dev/cassandra_record/issues/2

### CODE REVIEW

File: spec/integration/cassandra_record/base_spec.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'spec_helper'
+require 'pry'
```
Comment (arielvalentin) [inline spec/integration/cassandra_record/base_spec.rb]: nitpick

File: spec/integration/cassandra_record/base_spec.rb [hunk 2]
```diff
@@ -103,25 +104,63 @@ def create
...
+        insert_4,
+        insert_5
+      ]
+    end
+
+    before do
+      inserts.each do |record|
+        insert_statement = <<-CQL
+          INSERT INTO #{keyspace}.test_records (id, name)
+          VALUES (#{record[0]}, '#{record[1]}');
```
Comment (arielvalentin) [inline spec/integration/cassandra_record/base_spec.rb:126]: I don't know cassandra much. Is there a concept like Prepared Statements?

File: lib/cassandra_record/base.rb [hunk 3]
```diff
@@ -20,6 +20,24 @@ def batch_create(array_of_attributes, options={})
...
       end
 
+      # combines results from all pages
+      # into a single result set
+      def all(attributes={})
+        paginated_result_set = where(attributes)
+        rows = []
+
+        loop do
+          paginated_result_set.each do |row|
```
Comment (arielvalentin) [inline lib/cassandra_record/base.rb:30]: Is this a cassandra API quirk? Does it not implement enumerable?

Comment (toddmohney) [inline lib/cassandra_record/base.rb:30]: The underlying gem returns an Enumerable `Result`, but that `Result` is actually paginated, thus the bug with our implementation.

My intention here is to leave the `.where` method as is to allow clients to paginate however they'd like (read non-breaking API change), but add the `.all` method for clients that would prefer to treat the results as a complete, unpaginated set.

File: spec/integration/cassandra_record/base_spec.rb (hunk 2 above)
Comment (toddmohney) [inline spec/integration/cassandra_record/base_spec.rb:126]: Yes, but this is test code. I'm not super concerned about optimizing it right now.

**Pull request merged**